pkg/gatewayserver/io/mqtt/mqtt.go

```diff
@@ -72,8 +73,18 @@ func (s *srv) accept() error {
 			return err
 		}
 
+		remoteAddr := mqttConn.RemoteAddr().String()
```

Comment:
Are we sure this does what we think it does ? `RemoteAddr` contains the port as well, and as far as I know the port will differ on each connection attempt. Also this is open to collisions due to reverse-proxies.

Reply:
The address port is stripped in `ratelimit.GatewayAcceptMQTTConnectionResource`.

> Also this is open to collisions due to reverse-proxies.

On the client side, we cannot really do much for this (since for example all gateways behind a NAT will always have the same remote IP address). 

If MQTT (or UDP) endpoints are served behind a reverse proxy, then indeed this will not work. I believe TTN and TTSC are not affected by this, since the endpoints are not behind a reverse proxy.

It is definitely worth adding a note for this on the documentation.

cc @htdvisser

Reply:
As I already wrote to @neoaggelos, we should be able to configure "trusted" IP addresses that bypass rate limiting for such situations. 

For now, we should just document that rate limiting must be done by the proxy (and disabled in The Things Stack) if a reverse proxy is used.

In the future we may support PROXY protocol to get the real client IP from load balancers that support it.

Reply:
Perhaps a better solution would be to take the gateway ID into consideration for the rate limiting key? An issue would be that the MQTT frontend does too many things before terminating the connection in that case.

Reply:
The gateway ID is not really available at this level (this is the TCP accept routine, what is 'one the wire' is still unknown at this layer.

I'm ok with the trusted source IPs + documentation, and agree that there isn't much that can be done at this layer.